pkg/schema/error.go

```diff
@@ -4,25 +4,47 @@
 package schema
 
 import (
+	"bytes"
 	"fmt"
+	"log"
 	"strings"
+	"text/template"
 
 	"github.com/k14s/ytt/pkg/filepos"
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
-func NewInvalidSchemaError(found yamlmeta.Node, message, hint string) error {
-	return &invalidSchemaError{
-		Message: message,
-		Found:   found,
-		Hint:    hint,
+const invalidSchemaTemplate = `
+{{.Title}}
+
+{{.FileName}}:
+{{pad "|" false}}
+{{pad "|" true}} {{.Diff}}
+{{pad "|" false}}
+
+{{with .Found}}{{pad "=" false}} found: {{.}}{{end}}
+{{with .Expected}}{{pad "=" false}} expected: {{.}}{{end}}
+{{range .Hints}}{{pad "=" false}} hint: {{.}}
+{{end}}`
```

Comment:
I dig it:
- 👍🏻  delegate formatting to stdlib
- 👍🏻  declarative, concise, and easy to edit.